### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by silkemarie**
removed unused Iterator and Union from imports, because mypy told me my code would fail if it didn't get string input to put in upper() method, so now I'm always asking for string input as well. 
Also validation on student id being unique! in new validator 😸

### CODE REVIEW

File: api/main.py [hunk 1]
```diff
@@ -16,41 +19,56 @@ def student_id_positive(cls, value):
             raise ValueError(f"Expected positive price, received {value}")
         return value
 
+    @validator("student_id")
+    def student_id_unique(cls, value):
+        if value in [student["student_id"] for student in students]:
```
Comment (shahnozaAI) [inline api/main.py:24]: :scream_cat: :scream_cat: beautiful :fire:

Comment (silkemarie) [reply api/main.py:24]: ty!! 🥰

File: api/main.py [hunk 2]
```diff
@@ -16,41 +19,56 @@ def student_id_positive(cls, value):
...
     return students
 
+
 @app.delete("/students/{student_id}")
 def delete_student(student_id: int):
     for i in range(len(students)):
         if students[i]["student_id"] == student_id:
             del students[i]
             break
     print(students)
```
Comment (shahnozaAI) [inline api/main.py:73]: prints are only for development, we avoid them in production --> instead we use loggers
I would delete it :)

**Review by shahnozaAI — APPROVED** [review]
:dancing_women:

File: api/main.py (hunk 2 above)
Comment (silkemarie) [reply api/main.py:73]: I'll delete it! :)

**Pull request merged**